extensions/amp-lightbox-gallery/1.0/README.md

```diff
@@ -0,0 +1,274 @@
+       lightbox
+     />
+     <figcaption>dog wearing yellow shirt.</figcaption>
+    </figure>
+  <div class="buttons" style="margin-top: 8px;">
+    <button id="change-img">
+      change image
+    </button>
+  </div>
+
+  <script>
+    (async () => {
+      const img = document.queryselector('#my-img');
+      await customelements.whendefined('img');
+      // set up button actions
+      document.queryselector('#change-img').onclick = () => {
+        img.setattribute('src', 'https://images.unsplash.com/photo-1603123853880-a92fafb7809f?ixlib=rb-1.2.1&auto=format&fit=crop&w=1498&q=80')
+      }
+    })();
+  </script>
+```
+
+[/example]
+
+### Usage
+
+To use `bento-liightbox-gallery`, ensure the required script is included in your `<head>` section,
+then add the `lightbox` attribute on an `<img>` or `<bento-carousel>` element.
+
+#### Add Captions
+
+Optionally, you can specify a caption for each element in the lightbox. these
+fields are automatically read and displayed by the `<bento-lightbox-gallery>` in
+the following order of priority:
+
+-   `figcaption` (if the lightboxed element is the child of a figure)
+-   `aria-describedby`
+-   `alt`
+-   `aria-label`
+-   `aria-labelledby`
+
+In the following example, `<bento-lightbox-gallery>` displays the `figcaption`
+value as its description, showing "toront's cn tower was ....".
+
+```html
+<figure>
+  <img
+    id="hero-img"
+    lightbox="toronto"
+    src="https://picsum.photos/1600/900?image=1075"
+    layout="responsive"
+    width="1600"
+    height="900"
+    alt="picture of cn tower."
+  >
+  <figcaption class="image">
+    toronto's cn tower was built in 1976 and was the tallest free-standing
+    structure until 2007.
+  </figcaption>
+</figure>
+```
+
+In the following example, `<bento-lightbox-gallery>` displays the `alt` value as
+its description, showing "picture of cn tower".
+
+```html
+<img
+  id="hero-img"
+  lightbox="toronto"
+  src="https://picsum.photos/1600/900?image=1075"
+  layout="responsive"
+  width="1600"
+  height="900"
+  alt="picture of cn tower"
+>
+```
+
+#### Implement Thumbnail Previews
+
+Lightboxed items have a thumbnail gallery view. you can optionally specify a
+thumbnail item for your lightboxed element via the attribute
+`lightbox-thumbnail-id` that references the `id` of an `<img>` element with
+`layout="nodisplay"`.
+
+```html
+<bento-youtube
+  width="480"
+  height="270"
+  layout="responsive"
+  data-videoid="lbtcb7yls8y"
+  lightbox-thumbnail-id="my-thumbnail-img"
+>
+</bento-youtube>
+
+<img
+  id="my-thumbnail-img"
+  width="200"
+  height="200"
+  layout="nodisplay"
+  src="https://picsum.photos/200/200?image=1074"
+>
+```
+
+If no thumbnail is specified, `<img>` elements will be cropped per
+`object-fit: cover`, `<bento-video>` will use the image `src` specified in its
+`poster` attribute, and placeholder images will be used for lightboxed elements
+that have one.
```

Comment:
This section can be removed since the `lightbox-thumbnail-id` is not supported at this time, see #32759